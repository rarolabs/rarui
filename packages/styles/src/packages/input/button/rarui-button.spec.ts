import { button } from ".";

describe("Button Styles", () => {
  it("should apply base styles correctly", () => {
    const className = button.classnames.button({
      size: "large",
      appearance: "brand",
      variant: "solid",
    });
    expect(className).toContain("button");
  });

  it("should apply full width variant styles", () => {
    const className = button.classnames.button({ full: true });
    expect(className).toContain("full");
    expect(className).toMatchSnapshot("full-width");
  });

  it("should apply appearance variants correctly", () => {
    const appearances = [
      "brand",
      "danger",
      "success",
      "warning",
      "neutral",
      "inverted",
    ] as const;
    appearances.forEach((appearance) => {
      const className = button.classnames.button({
        appearance,
        variant: "solid",
      });
      expect(className).toContain(appearance);
      expect(className).toMatchSnapshot(`appearance-${appearance}`);
    });
  });

  it("should apply size variants correctly", () => {
    const sizes = ["large", "medium", "small"] as const;
    sizes.forEach((size) => {
      const className = button.classnames.button({
        size,
        appearance: "brand",
        variant: "solid",
      });
      expect(className).toContain(size);
      expect(className).toMatchSnapshot(`size-${size}`);
    });
  });

  it("should apply variant styles correctly", () => {
    const variants = ["solid", "outlined", "text", "tonal"] as const;
    variants.forEach((variant) => {
      const className = button.classnames.button({
        appearance: "brand",
        variant,
      });
      expect(className).toContain(variant);
      expect(className).toMatchSnapshot(`variant-${variant}`);
    });
  });

  it("should apply compound variants correctly", () => {
    const compoundVariants = [
      { appearance: "brand", variant: "outlined" },
      { appearance: "danger", variant: "tonal" },
    ] as const;

    compoundVariants.forEach(({ appearance, variant }) => {
      const className = button.classnames.button({ appearance, variant });
      expect(className).toContain(appearance);
      expect(className).toContain(variant);
      expect(className).toMatchSnapshot(
        `compound-variant-${appearance}-${variant}`,
      );
    });
  });

  it("should apply pseudo-class styles correctly", () => {
    const className = button.classnames.button({
      appearance: "brand",
      variant: "solid",
    });
    expect(className).toMatchSnapshot("pseudo-classes");
  });
});
