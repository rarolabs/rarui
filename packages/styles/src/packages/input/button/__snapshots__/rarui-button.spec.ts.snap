// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Styles should apply appearance variants correctly: appearance-brand 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply appearance variants correctly: appearance-danger 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_danger__15gxywr3 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply appearance variants correctly: appearance-inverted 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_inverted__15gxywr7 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply appearance variants correctly: appearance-neutral 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_neutral__15gxywr6 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply appearance variants correctly: appearance-success 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_success__15gxywr4 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply appearance variants correctly: appearance-warning 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_warning__15gxywr5 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply compound variants correctly: compound-variant-brand-outlined 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_outlined__15gxywrc rarui-button__compound_0__15gxywrf"`;

exports[`Button Styles should apply compound variants correctly: compound-variant-danger-tonal 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_danger__15gxywr3 rarui-button_variant_tonal__15gxywre rarui-button__compound_13__15gxywrs"`;

exports[`Button Styles should apply full width variant styles: full-width 1`] = `"rarui-button__15gxywr0 rarui-button_full_true__15gxywr1"`;

exports[`Button Styles should apply pseudo-class styles correctly: pseudo-classes 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply size variants correctly: size-large 1`] = `"rarui-button__15gxywr0 rarui-button_size_large__15gxywr8 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply size variants correctly: size-medium 1`] = `"rarui-button__15gxywr0 rarui-button_size_medium__15gxywr9 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply size variants correctly: size-small 1`] = `"rarui-button__15gxywr0 rarui-button_size_small__15gxywra rarui-button_appearance_brand__15gxywr2 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply variant styles correctly: variant-outlined 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_outlined__15gxywrc rarui-button__compound_0__15gxywrf"`;

exports[`Button Styles should apply variant styles correctly: variant-solid 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_solid__15gxywrb"`;

exports[`Button Styles should apply variant styles correctly: variant-text 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_text__15gxywrd rarui-button__compound_6__15gxywrl"`;

exports[`Button Styles should apply variant styles correctly: variant-tonal 1`] = `"rarui-button__15gxywr0 rarui-button_appearance_brand__15gxywr2 rarui-button_variant_tonal__15gxywre rarui-button__compound_12__15gxywrr"`;
